.github/workflows/publish.yaml

```diff
@@ -0,0 +1,42 @@
+name: Publish
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  charm:
+    name: Charm
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Install dependencies
+      run: |
+        set -eux
+        sudo snap install charm --classic
+        sudo pip3 install charmcraft
+
+    - name: Publish charm
+      env:
+        CHARMSTORE_CREDENTIAL: ${{ secrets.CHARMSTORE_CREDENTIAL }}
+      run: |
+        set -eux
+        echo $CHARMSTORE_CREDENTIAL > ~/.go-cookies
+
+        CS_URL="cs:~kubeflow-charmers/kubeflow-dashboard"
+
+        IMAGE=$(yq eval '.resources.oci-image.upstream-source' metadata.yaml)
```

Comment:
You'll need to snap install `yq`. See here for my same copy/paste error: https://github.com/canonical/tfjob-operator/pull/2